src/Group.cpp

```diff
@@ -0,0 +1,195 @@
+#include "global.h"
+#include "SongManager.h"
+#include "RageFile.h"
+#include "RageUtil.h"
+#include "RageFileManager.h"
+#include "RageLog.h"
+#include "Song.h"
+#include "SongCacheIndex.h"
+#include "SongUtil.h"
+#include "TitleSubstitution.h"
+#include "Group.h"
+
+#include <cstddef>
+#include <tuple>
+#include <vector>
+
+
+/** @brief The file that contains the group information.
+ * We name this Pack.ini over Group.ini to avoid conflict
+ * with the Waiei Group.ini-lua project still actively
+ * being developed.
+*/
+const RString INI_FILE = "Pack.ini";
+const int INI_VERSION = 1;
+
+Group::Group() {
+    m_sDisplayTitle = "";
+    m_sSortTitle = "";
+    m_sPath = "";
+    m_sGroupName = "";
+    m_sTranslitTitle = "";
+    m_sSeries = "";
+    m_fSyncOffset = 0;
+    m_bHasPackIni = false;
+    m_iYearReleased = 0;
+    m_sBannerPath = "";
+}
+
+Group::~Group() {
+    SONGMAN->GetGroupGroupMap().erase(m_sGroupName);
+}
+
+Group::Group(const RString& sDir, const RString& sGroupDirName) {
+    RString sPackIniPath = sDir + sGroupDirName + "/" + INI_FILE;
+    m_sPath = sDir + sGroupDirName;
+    m_sGroupName = Basename(sGroupDirName);
+    m_sDisplayTitle = m_sGroupName;
+    m_sSortTitle = m_sGroupName;
+    m_sTranslitTitle = m_sGroupName;
+    m_sSeries = "";
+    m_fSyncOffset = PREFSMAN->m_fMachineSyncBias;
+    m_bHasPackIni = false;
+    m_iYearReleased = 0;
+    m_sBannerPath = "";
+    m_iVersion = INI_VERSION;
+    
+    if (FILEMAN->DoesFileExist(sPackIniPath)) {
+        IniFile ini;
+        ini.ReadFile(sPackIniPath);
+
+        RString sVersion = "";
+        ini.GetValue("Group", "Version", sVersion);
+        Trim(sVersion);
+
+        // Only read the Pack.ini if the version is set
+        // Otherwise log a warning and skip reading the 
+        if (!sVersion.empty()) {
+            m_bHasPackIni = true;
+            m_iVersion = StringToInt(sVersion);
+
+            // Define a vector of key-value pairs to cleanly iterate
+            std::vector<std::pair<RString, RString&>> vPackfields = {
+                {"DisplayTitle", m_sDisplayTitle},
+                {"SortTitle", m_sSortTitle},
+                {"TranslitTitle", m_sTranslitTitle},
+                {"Series", m_sSeries},
+                {"Banner", m_sBannerPath}
+            };
+
+            for (auto& [key, value] : vPackfields) {
+                ini.GetValue("Group", key, value);
+                Trim(value);
+
+                if (value.empty()) {
+                    if (key == "DisplayTitle" || key == "SortTitle" ) {
+                        value = m_sGroupName;
+                    } else if (key == "TranslitTitle") {
+                        value = m_sDisplayTitle;
+                    } 
+                }
+            }
+            
+            RString sValue = "";
+            ini.GetValue("Group", "SyncOffset", sValue);
+            Trim(sValue);
+            if (!sValue.empty()) {
+                if (sValue.CompareNoCase("null") == 0) {
```

Comment:
Let's just standardize on it explicitly being capitalized (ITG / NULL). I think it's kinda meh if it's mixed and matched across pack.ini files.